### PR TITLE
Gml 1207 add more metrics

### PR DESCRIPTION
**Pull request by parkererickson-tg**

### CODE REVIEW

**Review by qe-tigergraph — COMMENTED** [review]
Unit Test: SUCCESS, e2e Test: SUCCESS, Jenkins_job:http://192.168.99.101:30080/job/mlwb_build/714/

**Review by qe-tigergraph — APPROVED** [review]
QE Approved

File: pyTigerGraph/gds/metrics.py [hunk 1]
```diff
@@ -199,7 +323,7 @@ def update(self, preds: ndarray, labels: ndarray) -> None:
             labels
         ), "The lists of predictions and labels must have same length"
         self._cumsum += float(((preds - labels)**2).sum())
-        self._count += int(preds.sum())
+        self._count += int(len(preds))
```
Comment (billshitg) [inline pyTigerGraph/gds/metrics.py:326]: Just nitpicking: len() is already int

File: pyTigerGraph/gds/metrics.py [hunk 2]
```diff
@@ -274,6 +400,98 @@ def value(self) -> float:
...
+    Also known as Precsion@K.
+
+    Usage:
+
+    * Call the update function to add predictions and labels.
+    * Get Hits@K value at any point by accessing the value property.
+
+    Args:
+        k (int):
+            Number of classes in your classification task.
```
Comment (billshitg) [inline pyTigerGraph/gds/metrics.py:417]: Is k the number of classes?

File: pyTigerGraph/gds/metrics.py [hunk 3]
```diff
@@ -274,6 +400,98 @@ def value(self) -> float:
...
+    This metric is used in link prediction tasks, i.e. determining if two vertices have an edge between them
+
+    Usage:
+
+    * Call the update function to add predictions and labels.
+    * Get Recall@K value at any point by accessing the value property.
+
+    Args:
+        k (int):
+            Number of classes in your classification task.
```
Comment (billshitg) [inline pyTigerGraph/gds/metrics.py:463]: Same here

File: tests/test_gds_metrics.py [hunk 4]
```diff
@@ -75,6 +86,135 @@ def test_update(self):
...
+        measure.update(preds, truth)
+        self.assertEqual(measure.value, 1/3)
+
+class TestGDSConfusionMatrix(unittest.TestCase):
+    def test_init(self):
+        measure = ConfusionMatrix(num_classes=2)
+        self.assertIsNone(measure.value)
+
+    def test_update(self):
+        measure = ConfusionMatrix(num_classes=2)
```
Comment (billshitg) [inline tests/test_gds_metrics.py:209]: Do we have test for num_classes>2 for any of the the ConfusionMatrix related metrics?

**Review by billshitg — APPROVED** [review]
LGTM. There are some nitpicking issues but they don't affect the functioning of the code

**Review by billshitg — APPROVED** [review]
thanks for resolving the minor comments. look good to me

**Review by qe-tigergraph — CHANGES_REQUESTED** [review]
Unit Test: FAILURE, Jenkins_job:http://192.168.99.101:30080/job/mlwb_build/717/

**Review by qe-tigergraph — CHANGES_REQUESTED** [review]
Unit Test: FAILURE, Jenkins_job:http://192.168.99.101:30080/job/mlwb_build/718/

**Review by qe-tigergraph — CHANGES_REQUESTED** [review]
Unit Test: FAILURE, Jenkins_job:http://192.168.99.101:30080/job/mlwb_build/719/

**Review by qe-tigergraph — CHANGES_REQUESTED** [review]
Unit Test: FAILURE, Jenkins_job:http://192.168.99.101:30080/job/mlwb_build/720/

**Review by qe-tigergraph — CHANGES_REQUESTED** [review]
Unit Test: FAILURE, Jenkins_job:http://192.168.99.101:30080/job/mlwb_build/721/

**Review by qe-tigergraph — CHANGES_REQUESTED** [review]
Unit Test: SUCCESS, e2e Test: FAILURE, Jenkins_job:http://192.168.99.101:30080/job/mlwb_build/723/

**Review by qe-tigergraph — COMMENTED** [review]
Unit Test: SUCCESS, e2e Test: SUCCESS, Jenkins_job:http://192.168.99.101:30080/job/mlwb_build/724/

**Review by qe-tigergraph — APPROVED** [review]
QE Approved

**Pull request merged**